server/channels/app/preference.go

```diff
@@ -20,27 +22,27 @@ type preferencesServiceWrapper struct {
 	app AppIface
 }
 
-func (w *preferencesServiceWrapper) GetPreferencesForUser(userID string) (model.Preferences, *model.AppError) {
-	return w.app.GetPreferencesForUser(userID)
+func (w *preferencesServiceWrapper) GetPreferencesForUser(c request.CTX, userID string) (model.Preferences, *model.AppError) {
+	return w.app.GetPreferencesForUser(c, userID)
 }
 
-func (w *preferencesServiceWrapper) UpdatePreferencesForUser(userID string, preferences model.Preferences) *model.AppError {
-	return w.app.UpdatePreferences(userID, preferences)
+func (w *preferencesServiceWrapper) UpdatePreferencesForUser(c request.CTX, userID string, preferences model.Preferences) *model.AppError {
+	return w.app.UpdatePreferences(c, userID, preferences)
 }
 
-func (w *preferencesServiceWrapper) DeletePreferencesForUser(userID string, preferences model.Preferences) *model.AppError {
-	return w.app.DeletePreferences(userID, preferences)
+func (w *preferencesServiceWrapper) DeletePreferencesForUser(c request.CTX, userID string, preferences model.Preferences) *model.AppError {
+	return w.app.DeletePreferences(c, userID, preferences)
 }
 
-func (a *App) GetPreferencesForUser(userID string) (model.Preferences, *model.AppError) {
+func (a *App) GetPreferencesForUser(c request.CTX, userID string) (model.Preferences, *model.AppError) {
```

Comment:
Most of these changes are because I needed the app context to be able to call a plugin hook. I only needed it in `UpdatePreferences`, but I figured I'd add it to all of the methods in `app/preferences.go` for completion's sake